src/Stratis.Bitcoin.Features.Wallet/WalletManager.cs

```diff
@@ -251,7 +251,11 @@ public void Start()
             // the next WalletSyncManager.OrchestrateWalletSync invocation.
             foreach (string walletName in this.WalletRepository.GetWalletNames())
             {
-                this.WalletRepository.RewindWallet(walletName, this.ChainIndexer.Tip);
```

Comment:
This will leave the wallet partially rewound even when its completely on a different chain.